### DISCUSSION TITLE
[Poll] Which feature do you need most?

### DISCUSSION THREAD

**Discussion by Asm-Def** [Kanaries/pygwalker#72] [Announcements]
:mega: Hello, community!

We've accumulated a ton of feature requests:sparkles: in recent weeks, some from "issues" and some from maintainers.

We're interested in taking a look at the **popularity** of these requests so that we can meet broader needs in a more timely manner.

In this discussion we will bring up all the feature requests we will be  :construction:developing in a few weeks, and you are welcome to **upvote** for your most desired features. Features with more votes will be developed with higher priority.

You can also :link:link your feature request in a comment or :bulb:add your thoughts as replies to comments.

For more real-time replies, please join our discord community
<a href="https://discord.gg/Z4ngFWXz2U"><img alt="discord invitation link" src="https://dcbadge.vercel.app/api/server/Z4ngFWXz2U?style=flat" align="center"></a>

**Comment by Asm-Def** [comment]
[Feat] Add support for streamlit
#11

**Reply by Asm-Def** [reply]
Now we have Streamlit initially supported in pr #74 .
Please update to `pygwalker>=0.1.4.9` for trial.

**Comment by Asm-Def** [comment]
[Feat] Add support for Plotly dash (#16 )

**Comment by Asm-Def** [comment]
[Feat] Support Large-Scaled data, maybe with PySpark (#18 , #62 )

**Comment by Asm-Def** [comment]
[Feat] Add support for Panel Holoviz (already achieved by developers of Panel and will be released in v1.0) #37

**Comment by Asm-Def** [comment]
[Feat] Preload charts or pre-set data types in code, no need to manually reconfigure each time (#60 , #69, #70)

**Comment by Asm-Def** [comment]
[Feat] Add support for zeppelin notebook (#62)

**Comment by Asm-Def** [comment]
[Feat] Some enhancements to static charts (like in Excel) (#64)

**Comment by Asm-Def** [comment]
[Feat] Crosstab to excel (#69 )

**Comment by Asm-Def** [comment]
[Feat] Datetime Aggregation (#58 )

**Comment by hucker** [comment]
It would be nice to have access to controlling the axis ranges.  I use a tool called JMP.  They let you drag the axis range with the mouse by dragging the axis itself.  This is very intuitive.  Same with scaling, you put the mouse on a value on the axis and hit the scroll wheel, and it scales that axis, no typing in ranges or needing UI with lots of min/max/range data entry.

**Reply by Asm-Def** [reply]
Actually, controlling the axis ranges is already supported if you check the "Axis Resizing" box, but the label might be kind of confusing.
![img](https://user-images.githubusercontent.com/8137814/224220115-50e66bea-e78f-427c-802d-aa4211b67305.png)

**Comment by Asm-Def** [comment]
[Feat] Reduce unnecessary storage consumption, both memory and disk (#76 )

**Comment by sailingwave** [comment]
This is a great vis tool! Please keep up the great work.

It'd be nice to be able to customize the color.
